### PR TITLE
Tl/add ninja build to BuildRequires

### PR DESCRIPTION
**Pull request by tonylambiris**
**Summary of changes**:
Add the `ninja-build` package to cilium.spec.envsubst (required for recent builds, verified on RHEL and Arch Linux)
Fix-up the VERSION variable.
Generate a RELEASE variable (previously hard-coded to 0).

Here is the current package information after building:
```
➜  rpm -qip output/cilium-1.2.90-0.git54ce03f.fc28.x86_64.rpm | head -3
Name        : cilium
Version     : 1.2.90
Release     : 0.git54ce03f.fc28
```

Here is the updated package information from this PR:
```
➜  rpm -qip output/cilium-1.2.0.rc3-1340.git562985b.fc28.x86_64.rpm | head -3 
Name        : cilium
Version     : 1.2.0.rc3
Release     : 1340.git562985b.fc28
```

**How to test (optional)**:
```bash
$ cd contrib/packaging/rpm && make
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5220)

### CODE REVIEW

**Comment by ianvernon** [comment]
Thank you for the contributions @tonylambiris ! Can you please remove the "Merge" commit from your branch by rebasing, as well as sign off each commit per the [contributing documentation](http://docs.cilium.io/en/v1.1/contributing/#developer-s-certificate-of-origin) ?

**Review by joestringer — CHANGES_REQUESTED** [review]
Looks good to me, just needs rebase/squash and signoff per @ianvernon 's comments.

Thanks!

**Comment by tonylambiris** [comment]
@ianvernon @joestringer Hey guys, did I do this correctly? Haven't done much in the way of squashing and rebasing commits.

**Comment by joestringer** [comment]
@tonylambiris Hey, I think it didn't quite work.

There's a guide on how to rebase here:
https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history

I usually do something similar to the section about "Interactive Rebase" and "Squash the commits together".

Then, to sign off, assuming you agree with the [Contributing guidelines](http://docs.cilium.io/en/v1.1/contributing/#developer-s-certificate-of-origin), you can do a `git commit --amend -s` and save the commit message.

File: contrib/packaging/rpm/Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 include ../../../Makefile.defs
 
+VERSION := $(shell git tag -l --sort=-v:refname | sed 's/v\([^-]*-g\)/r\1/;s/-/./g' | head -1)
```
Comment (aanm) [inline contrib/packaging/rpm/Makefile]: This should be available from `./VERSION`. Did we change anything that broke it?

**Comment by aanm** [comment]
@tonylambiris try `git rebase -i ^HEAD~8` and replace `pick` with `squash` or `s` if you want to squash commits together or simply remove the lines for the commits that you want to make them disappear.

The do `git fetch origin master` and `git rebase origin/master`

**Comment by tonylambiris** [comment]
@aanm did that work properly? if not im at a loss because I had rebased and squashed. running `git rebase -i` currently shows:
```noop

# Rebase 562985b8c..562985b8c onto 562985b8c (1 command)
```

let me know, i'll just redo the pull request. what exactly was the original issue anyway here (so I can avoid it in the future).

thanks.

**Comment by aanm** [comment]
@tonylambiris are you on cilium slack? I can help you quickly in there

**Comment by tgraf** [comment]
@tonylambiris `git rebase -i master`, replace pick with squash except for first line, `git commit --amend -s` to sign the commit, `git push -f`

**Comment by raybejjani** [comment]
If we prefer saving tonylambiris the hassle, perhaps an admin can click the "Squash and merge" button? I think that will work correctly in this case (I'm not sure, though).

**Comment by joestringer** [comment]
> If we prefer saving tonylambiris the hassle, perhaps an admin can click the "Squash and merge" button? I think that will work correctly in this case (I'm not sure, though).

This might just work, but we still need sign-off from @tonylambiris.

**Comment by tonylambiris** [comment]
Is it copisetic if I close this PR in favor of a new one, all refreshly rebased and such? Caught a one offs I need to add (ie: requirement of llvm installed due to llc and/or lld needed for pre-flight checks).

Thanks!

**Comment by tgraf** [comment]
> Is it copisetic if I close this PR in favor of a new one, all refreshly rebased and such? Caught a one offs I need to add (ie: requirement of llvm installed due to llc and/or lld needed for pre-flight checks).
> 
> Thanks!

Sure, go ahead

**Comment by tonylambiris** [comment]
I'll get back to this shortly, thanks everyone for their input! 👍

**Pull request closed without merging**